evalml/tests/data_checks_tests/test_natural_language_nan_data_check.py

```diff
@@ -0,0 +1,112 @@
+    assert nl_nan_check.validate(data) == {
+        "warnings": [],
+        "actions": [],
+        "errors": []
+    }
+
+
+def test_nl_nan_data_check_error_multiple_nl_nan():
+    data = pd.DataFrame()
+    data['A'] = pd.Series([None, "string_that_is_long_enough_for_natural_language", "string_that_is_long_enough_for_natural_language"])
+    data['B'] = pd.Series([None, "string_that_is_long_enough_for_natural_language", "string_that_is_long_enough_for_natural_language"])
+    data['C'] = np.random.randint(0, 5, size=len(data))
+
+    nl_nan_check = NaturalLanguageNaNDataCheck()
+    assert nl_nan_check.validate(data) == {
+        "warnings": [],
+        "actions": [],
+        "errors": [DataCheckError(message='Input natural language column(s) (A, B) contains NaN values. Please impute NaN values or drop these rows or columns.',
+                                  data_check_name=NaturalLanguageNaNDataCheck.name,
+                                  message_code=DataCheckMessageCode.NATURAL_LANGUAGE_HAS_NAN,
+                                  details={"columns": 'A, B'}).to_dict()]
+    }
+
+
+def test_nl_nan_check_input_formats():
+    nl_nan_check = NaturalLanguageNaNDataCheck()
+
+    # test empty pd.DataFrame
+    assert nl_nan_check.validate(pd.DataFrame()) == {"warnings": [], "errors": [], "actions": []}
+
+    expected = {
+        "warnings": [],
+        "actions": [],
+        "errors": [DataCheckError(message='Input natural language column(s) (nl) contains NaN values. Please impute NaN values or drop these rows or columns.',
+                                  data_check_name=NaturalLanguageNaNDataCheck.name,
+                                  message_code=DataCheckMessageCode.NATURAL_LANGUAGE_HAS_NAN,
+                                  details={"columns": 'nl'}).to_dict()]
+    }
+
+    nl_col = [None, "string_that_is_long_enough_for_natural_language", "string_that_is_long_enough_for_natural_language"]
+
+    #  test Woodwork
+    ww_input = ww.DataTable(pd.DataFrame(nl_col, columns=['nl']))
+    assert nl_nan_check.validate(ww_input) == expected
+
+    expected = {
+        "warnings": [],
+        "actions": [],
+        "errors": [DataCheckError(message='Input natural language column(s) (0) contains NaN values. Please impute NaN values or drop these rows or columns.',
+                                  data_check_name=NaturalLanguageNaNDataCheck.name,
+                                  message_code=DataCheckMessageCode.NATURAL_LANGUAGE_HAS_NAN,
+                                  details={'columns': '0'}).to_dict()]
+    }
+
+    #  test 2D list
+    nl_col_without_nan = ["string_that_is_long_enough_for_natural_language", "string_that_is_long_enough_for_natural_language", "string_that_is_long_enough_for_natural_language"]
+    assert nl_nan_check.validate([nl_col, nl_col_without_nan]) == expected
+
+    # test np.array
+    assert nl_nan_check.validate(np.array([nl_col, nl_col_without_nan])) == expected
```

Comment:
Can we also add a test on this behavior on an empty string in a natural language column? 

```python
cols = ["", "string_that_is_long_enough_for_natural_language", "string_that_is_long_enough_for_natural_language"]
```

Reply:
@bchen1116 I believe an empty string should pass this check since the LSA component was only buging out out on `NaN`

`np.nan is an invalid document, expected byte or unicode string.` from #2000